### PR TITLE
supporting async now

### PR DESCRIPTION
**Pull request by egbertn**
I added 2 methods to support async fetching

### CODE REVIEW

File: EventbriteApiV3/.nuspec [hunk 1]
```diff
@@ -11,7 +11,7 @@
     <repository>https://github.com/DTONeill/EventbriteApiV3</repository>
     <requireLicenseAcceptance>false</requireLicenseAcceptance>
     <description>Wrapper in top of Eventbrite api v3. It allow you to fetch attendees and events easily using your application key.</description>
-    <releaseNotes>Initial 1.0.0 release</releaseNotes>
+    <releaseNotes>Initial 1.0.1 release</releaseNotes>
```
Comment (DTONeill) [inline EventbriteApiV3/.nuspec]: It's not the initial release anymore, you could remove "Initial"

File: EventbriteApiV3/Attendees/AttendeeSearchEventbriteRequest.cs [hunk 2]
```diff
@@ -9,13 +10,19 @@ public class AttendeeSearchEventbriteRequest : EventbriteRequestBase
...
-           
+
         }
 
         public AttendeeSearchApiResponse GetResponse()
         {
             var response = GetJsonResponse();
 
+            return JsonConvert.DeserializeObject<AttendeeSearchApiResponse>(response);
+        }
```
Comment (DTONeill) [inline EventbriteApiV3/Attendees/AttendeeSearchEventbriteRequest.cs:21]: Convention: Newline between methods

File: EventbriteApiV3/EventbriteContext.cs [hunk 3]
```diff
@@ -19,10 +21,24 @@ public EventsSearchApiResponse GetEvents(BaseSearchCriterias searchCriterias)
         {
             return new EventSearchEventbriteRequest(this, searchCriterias).GetResponse();
         }
+        public  Task< EventsSearchApiResponse> GetEventsAsync(BaseSearchCriterias searchCriterias)
+        {
+            return (new EventSearchEventbriteRequest(this, searchCriterias)).GetResponseAsync();
+        }
```
Comment (DTONeill) [inline EventbriteApiV3/EventbriteContext.cs:28]: new line here too.

File: EventbriteApiV3/EventbriteContext.cs [hunk 4]
```diff
@@ -19,10 +21,24 @@ public EventsSearchApiResponse GetEvents(BaseSearchCriterias searchCriterias)
...
+            return (new EventsOrganizationRequest(this, organisationId, searchCriterias)).GetResponseAsync();
 
+        }
         public AttendeeSearchApiResponse GetAttendees(double eventId, BaseSearchCriterias searchCriterias)
         {
             return new AttendeeSearchEventbriteRequest(this, eventId, searchCriterias).GetResponse();
         }
+
+
+		public Task< AttendeeSearchApiResponse> GetAttendeesAsync(double eventId, BaseSearchCriterias searchCriterias)
```
Comment (DTONeill) [inline EventbriteApiV3/EventbriteContext.cs]: remove tab before "public"

File: EventbriteApiV3/EventbriteRequestBase.cs [hunk 5]
```diff
@@ -49,5 +50,23 @@ protected string GetJsonResponse()
                 return sr.ReadToEnd();
             }
         }
+        protected async Task<string> GetJsonResponseAsync()
```
Comment (DTONeill) [inline EventbriteApiV3/EventbriteRequestBase.cs:53]: new line

File: EventbriteApiV3/EventbriteRequestBase.cs [hunk 6]
```diff
@@ -49,5 +50,23 @@ protected string GetJsonResponse()
...
+            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(Url);
+            request.Headers.Add("authorization", $"Bearer {Context.AppKey}");
+            var response = await request.GetResponseAsync();
+
+            using (Stream stream = response.GetResponseStream())
+            using (StreamReader sr = new StreamReader(stream))
+            {
+                return await sr.ReadToEndAsync();
+            }
+        }
```
Comment (DTONeill) [inline EventbriteApiV3/EventbriteRequestBase.cs:64]: new line

File: EventbriteApiV3/Events/EventsOrganizationRequest.cs [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+		: base(string.Format(Path, organizationId), context, criterias.ToNameValueCollection())
+		{
+
+		}
+
+		public async Task<EventsSearchApiResponse> GetResponseAsync()
+		{
+			var response = await GetStreamResponseAsync();
+			var JsonSerializer = new JsonSerializer();
+			var result = JsonSerializer.Deserialize< EventsSearchApiResponse>(new JsonTextReader( response ));
```
Comment (DTONeill) [inline EventbriteApiV3/Events/EventsOrganizationRequest.cs:19]: you could return result directly.

**Review by DTONeill — CHANGES_REQUESTED** [review]

**Comment by DTONeill** [comment]
Hi, i'm pretty happy to have a contributor for the project, are you an active user of the project?

**Comment by egbertn** [comment]
Hi Sure, I am.


------ Origineel bericht ------
Van: "DTONeill" <notifications@github.com>
Aan: "DTONeill/EventbriteApiV3" <EventbriteApiV3@noreply.github.com>
CC: "Egbert Nierop" <egbert_nierop@outlook.com>; "Author" 
<author@noreply.github.com>
Verzonden: 1-11-2020 17:51:26
Onderwerp: Re: [DTONeill/EventbriteApiV3] supporting async now (#2)

**Comment by egbertn** [comment]
I think I solved the issues. Let me know if all ok.

**Review by DTONeill — APPROVED** [review]

**Pull request merged**